diff --git a/setup.py b/setup.py
index fe2398d8..3031cd67 100644
--- a/setup.py
+++ b/setup.py
@@ -379,7 +379,7 @@ def try_cc(cc, cc_type):
         #    http://bugs.python.org/issue9031
         #    http://bugs.python.org/issue1222585
         # So just switch it manually and see if that works.
-        cmd = ['c++'] + lopt[cc_type] + [o_file.name,'-o',exe_file.name]
+        cmd = [os.environ.get('CXX', 'c++')] + lopt[cc_type] + [o_file.name,'-o',exe_file.name]
         p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
         lines = p.stdout.readlines()
         p.communicate()
